src/client/components/MainForm.tsx

```diff
@@ -73,11 +76,15 @@ export const MainForm = ({
     useState<UseRecaptchaReturnValue>();
 
   /**
-   * Executes the reCAPTCHA check.
+   * Executes the reCAPTCHA check and form submit tracking.
    * Prevents the form from submitting until the reCAPTCHA check is complete.
    */
   const handleSubmit = useCallback(
     (event: React.FormEvent<HTMLFormElement>) => {
+      if (formTrackingName) {
+        trackFormSubmit(formTrackingName);
```

Comment:
can we use data-link-name element attributes using click events, instead of calling ophan interaction events manually?
e.g.
![image](https://user-images.githubusercontent.com/29203769/147078313-f0d223a2-d988-49b4-8e9f-4d63cb0104bd.png)

Reply:
Is there any documentation on this? e.g. is it only for click events? does it work on form submit etc.

Reply:
I can't find documentation.
This is the implementation, so looks like it's just a click event listener... https://github.com/guardian/ophan/blob/ccaa57bcee3f5f3f83ec28973074c9b3f98f1438/tracker-js/assets/coffee/ophan/click-path-capture.coffee#L23

looks like you can also add `data-component` attribute to the parent element

Reply:
there are lots of examples of usage
https://github.com/search?q=org%3Aguardian+%22data-link-name%22&type=code

Reply:
Thanks! I can probably refactor it somewhat still, I'll switch to using the `data-link-name`. But keep the manual events for form submit/focus/blur.
For the social button tracking we can easily switch to the `data-link-names'

Reply:
ah I see, for form submit a user can hit enter key rather than press the button, so `data-link-name` probably won't pick up these events 🤔

Reply:
Yeah, using the `onSubmit` option picks up both clicking the submit button, and using enter to submit the form. The `data-link-name` only works on click.

So I didn't refactor anything around submit/focus/blur, and left them as interaction events (which were already being used in the onboarding flow for tracking).

I've changed the social buttons to use `data-link-name` as thats a good use case for it!